feat: add system prompt

munnam77 opened the pull request:
Completely revamped the getSystemPrompt function to include detailed sections for identity, design engineering manifesto, response workflow, design system, tech stack, code constitution, security charter, and example implementations.
Added new constants and imports for better modularity and future extensibility.
Updated the CONTINUE_PROMPT section to align with the new system prompt structure.

leex279 commented:
Hi @munnam77, add some description if this is a real PR and you want it to be considered to merge. Otherwise I will close it.

munnam77 commented:
Completely revamped the getSystemPrompt function to include detailed sections for identity, design engineering manifesto, response workflow, design system, tech stack, code constitution, security charter, and example implementations.
Added new constants and imports for better modularity and future extensibility.
Updated the CONTINUE_PROMPT section to align with the new system prompt structure.

munnam77 commented:
@leex279 Completely revamped the getSystemPrompt function to include detailed sections for identity, design engineering manifesto, response workflow, design system, tech stack, code constitution, security charter, and example implementations.
Added new constants and imports for better modularity and future extensibility.
Updated the CONTINUE_PROMPT section to align with the new system prompt structure.

leex279 commented:
Thanks, please write in the PR description as well :)

thecodacus commented:
looks like this PR updates the default system prompt. I would suggest you to add another prompt along side the default prompt. that way we will be able to revert to the original one and also can switch to this one for those who thin this works better

munnam77 commented:
Done

thecodacus commented on app/lib/.server/llm/prompts.ts:
this file is not required/used anymore please delete

thecodacus commented on app/lib/common/prompts/default_prompts.ts:
rename this file to `prompt.ts`

thecodacus commented on app/lib/common/prompts/optimized.ts:
no need to do any changes in this file.

thecodacus commented on app/lib/common/prompts/prompts.ts:
rename this file to something that you think appropriate according to the new prompt.
and add that in the `prompt-library.ts` file so that it can be selected in the setting window

thecodacus commented on wrangler.toml:
dont make any changes to this file as its not related to this PR

thecodacus commented on .tool-versions:
no need to make any changes to this also again not for this PR

thecodacus commented:
also I noticed that you are stared this PR from stable thats why we are seeing conflicts

i suggest you close this PR. 
* switch to main branch 
* pull the latest main branch
* then from there create a feature branch - advanced prompt 
* add the prompt file.. (only the new prompt file  and the updated `prompt-library.ts` file to register the new prompt)
* then create a new PR

that way we will only have clean PR
